andela/ah-scorpion-frontend
# #160440771 Article Favouriting

#### What does this PR do?
- Adds favouriting feature
#### Description of Task to be completed?
- Added a Favourite component
- Integrated the Favourite component with RenderArticle component
- Added actions to fetch data from the API
- Added unit tests
#### How should this be manually tested?
1. use `git` to `pull` from the branch `ft-favouriting-article-160440771`
2. ensure that your environment variables are in the format shown in the `.env-sample`

3.  run `source .env` to export your environment variables
4. run `npm start -s` to run the server
5. ensure you have a valid article `slug`. You can get one from here `https://authors-haven-api.herokuapp.com/` by using the `GET /api/v1/articles/ ` endpoint
6. on your browser add this address `localhost:3000/article/:slug`. Be sure to replace the `localhost:3000` if your server is different or running in a different port.
7. At the bottom of the article click on the `gray heart` icon
8. you should get a message that you need to login.
9. now use the `/login` url to login and repeat steps 6 and 7.
10. The colour of the `heart icon` should change to red and a message will be put below it that the article is now favourited.
11. on refreshing the page, the icon status should be still the same (red if favourited or grey in unfavourited)

#### What are the relevant pivotal tracker stories?
- [#160440771](https://www.pivotaltracker.com/story/show/160440771)
